src/plugins/workspace/public/utils.ts

```diff
@@ -297,15 +340,18 @@ export const fulfillRelatedConnections = (
 
 // Helper function to merge data sources with direct query connections
 export const mergeDataSourcesWithConnections = (
-  dataSources: DataSource[],
+  dataSources: DataSource[] | DataConnection[],
   directQueryConnections: DataSourceConnection[]
 ): DataSourceConnection[] => {
   const openSearchConnections = convertDataSourcesToOpenSearchConnections(dataSources);
-
-  return [
+  const dataConnections = convertDataSourcesToDataConnections(dataSources);
+  const result = [
     ...fulfillRelatedConnections(openSearchConnections, directQueryConnections),
     ...directQueryConnections,
-  ].sort((a, b) => a.name.localeCompare(b.name));
+    ...dataConnections,
+  ].sort((a, b) => a?.name?.localeCompare(b?.name));
```

Comment:
It doesn't seem optional chain is needed here since the type definition shows `name` is mandatory.

Reply:
Updated.